app/case/views/accounts/create.py

```diff
@@ -14,14 +17,23 @@
 @coordinator_or_better_required
 def account_detail_view(request):
     if request.method == "POST":
-        default_data = {"username": request.POST.get("email")}
-        data = merge_form_data(request.POST, default_data)
-        form = InviteParalegalForm(data)
-        if form.is_valid():
-            user = form.save()
-            return redirect("account-detail", user.pk)
-    else:
-        form = InviteParalegalForm()
+        email = request.POST.get("email")
+
+        # Is this User already in Clerk?
+        try:
+            existing_user = User.objects.get(email=email)
+        except:
+            existing_user = None
 
-    context = {"form": form}
-    return render(request, "case/accounts/create.html", context)
+        if existing_user and email.endswith("@anikalegal.com"):
+            async_task(set_up_new_user_task, existing_user.pk)
+            return redirect("account-detail", existing_user.pk)
+        else:
+            data = merge_form_data(request.POST, {"username": email})
+            form = InviteParalegalForm(data)
+            if form.is_valid():
+                user = form.save()
```

Comment:
I recommend you remove `set_up_new_user_task` from the `save` method of `InviteParalegalForm` so that the two invocations are in the same file, rather than spread out across two files. That was it is easier to see what is happening by looking at one file.